### PR TITLE
Add maven settings.xml with JBoss public repository configured

### PR DESCRIPTION
**Pull request by kwart**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -27,6 +27,9 @@ significantly, this can of course be reconsidered.
 Creaper uses Maven for build. Some build dependencies are only present in
 the JBoss.org Maven repository, so it's recommended to
 [configure Maven to use the JBoss Repository](https://developer.jboss.org/wiki/MavenGettingStarted-Developers).
+You can use custom `settings.xml` where the repository is configured already.
+The file is located in the project root folder and can be activated by
```
Comment (Ladicek) [inline CONTRIBUTING.md]: Folder is a Windows terminology. Please use "directory" :-)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -27,6 +27,9 @@ significantly, this can of course be reconsidered.
...
+providing `-s` maven parameter (`mvn -s settings.xml ...`).
```
Comment (Ladicek) [inline CONTRIBUTING.md]: Maven

**Comment by kwart** [comment]
Setting the `before_script` is not sufficient. Let's try sth. else. I'll update the commit with suggested fixes.

**Comment by Ladicek** [comment]
I'd like to get rid of `before_script` entirely. I'm not sure why it's required -- at least for Creaper, if the CI only did `mvn clean verify -s settings.xml`, that would be entirely sufficient.

**Comment by Ladicek** [comment]
Would `before_script: echo LOL` work? :-)

**Comment by kwart** [comment]
The correct phase is not `before_script` but `install`. And IMO the `echo LOL` should work. I'll change it.

**Comment by Ladicek** [comment]
I have no idea why the tests failed, but I found that we should probably do this instead of `echo LOL`: https://docs.travis-ci.com/user/customizing-the-build/#Skipping-the-Installation-Step

**Comment by kwart** [comment]
I like more the `echo LOL` than the boring `true`.

**Comment by kwart** [comment]
Trying `STANDARD` [ci evironment](https://docs.travis-ci.com/user/ci-environment/) as next step.

**Comment by Ladicek** [comment]
The standard env has IPv6 enabled, according to the docs, which might affect things a bit / a lot :-)

**Comment by Ladicek** [comment]
Passed, cool! BTW, what's `-Dmaven.test.redirectTestOutputToFile=true`? I guess it pollutes the console less, which is good, but is the file accessible somewhere in case something fails?

**Comment by kwart** [comment]
Yes, the [description](http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#redirectTestOutputToFile) says: Set this to "true" to redirect the unit test standard output to a file (found in reportsDirectory/testName-output.txt).

**Comment by kwart** [comment]
I'm thinking about few more `mvn` options `"-B -fae -q"` (batch, fail-at-end, quiet). WDYT?

**Comment by Ladicek** [comment]
I'm not sure what `-B` would bring us, but it can't hurt either. `-fae` sounds like a good idea.

`-q` -- what does it do? I'd like to have as much output available as possible in case of failures. I guess there's no chance I could possibly access the build workspace with Travis, so I'm not exactly a fan of reducing the console output.

**Comment by kwart** [comment]
The quiet output only show errors. But as far as we are under 4MB console output size, we are fine without this switch. So just adding `-B` and `-fae`.

**Comment by Ladicek** [comment]
Merged, thanks. I was thinking about doing this myself, but was too lazy :-)

**Pull request merged**